File: src/main/java/me/lokka30/levelledmobs/result/PlayerLevelSourceResult.java
package me.lokka30.levelledmobs.result;

/**
 * Used for PlayerLevelling options
 *
 * @author stumper66
 * @since 3.3.0
 */
public class PlayerLevelSourceResult {

    public PlayerLevelSourceResult(final int numericResult) {
        this.numericResult = numericResult;
        this.isNumericResult = true;
    }

    public PlayerLevelSourceResult(final String stringResult) {
        this.stringResult = stringResult != null ? stringResult : "";
        this.isNumericResult = false;
    }

    public int numericResult;
    public Integer randomVarianceResult;
    public String stringResult;
    public final boolean isNumericResult;
    public String homeNameUsed;
}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/result/PlayerLevelSourceResult.kt
package io.github.arcaneplugins.levelledmobs.result

/**
 * Used for PlayerLevelling options
 *
 * @author stumper66
 * @since 3.3.0
 */
class PlayerLevelSourceResult {
    var numericResult = 0f
    var randomVarianceResult: Float? = null
    var stringResult: String? = null
        private set
    var isNumericResult = false
        private set
    var homeNameUsed: String? = null

    constructor(numericResult: Float){
        this.numericResult = numericResult
        this.isNumericResult = true
    }

    constructor(stringResult: String?){
        this.stringResult = stringResult ?: ""
    }
}